encoding/kzg/prover/noicicle.go

```diff
@@ -0,0 +1,16 @@
+//go:build !icicle
+
+package prover
+
+import (
+	"fmt"
+
+	"github.com/Layr-Labs/eigenda/encoding"
+	"github.com/Layr-Labs/eigenda/encoding/fft"
+	"github.com/Layr-Labs/eigenda/encoding/kzg"
+)
+
+func CreateIcicleBackendProver(p *Prover, params encoding.EncodingParams, fs *fft.FFTSettings, ks *kzg.KZGSettings) (*ParametrizedProver, error) {
+	// Not supported
+	return nil, fmt.Errorf("icicle backend called without icicle build tag")
```

Comment:
The point of noicicle.go file is that the icicle backend should only be functional when the icicle build tag is used. This is so we don't need to include the icicle backend files in the traditional encoder backend, otherwise it may break the existing build.